paasta_tools/contrib/kill_bad_containers.py

```diff
@@ -0,0 +1,78 @@
+#!/opt/venvs/paasta-tools/bin/python3
+import sys
+
+import iptc
+
+from paasta_tools import iptables
+from paasta_tools.utils import get_docker_client
+
+
+def get_container_from_dport(dport, docker_client):
+    for container in docker_client.containers():
+        try:
+            ports = container['Ports']
+            for port in ports:
+                if "PublicPort" in port:
+                    if port["PublicPort"] == int(dport):
+                        return container
+        except KeyError:
+            print(ports)
```

Comment:
this feels too defensive or maybe i misunderstand something, if KeyError is thrown, it means `Ports` is not found in `container` and `ports` variable will be `None`, what information do you  expect to get out of that `print` call?

how about:

    for container in docker_client.containers():
      for ports in container.get('Ports', []):
        for port in ports:
          if port.get('PublicPort') == int(dport):
            return container

Reply:
Hah. It is not too defensive. The world is a zoo and I found all sorts of weird things.